Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/tracing/details/TracingDetailsState.kt

```diff
@@ -56,48 +57,38 @@ data class TracingDetailsState(
         val resources = c.resources
         val days = daysSinceLastExposure
         val count = matchedKeyCount
-        return when (riskLevelScore) {
-            RiskLevelConstants.INCREASED_RISK ->
-                resources.getQuantityString(
-                    R.plurals.risk_details_information_body_increased_risk,
-                    days,
-                    days
-                )
-            RiskLevelConstants.UNKNOWN_RISK_OUTDATED_RESULTS ->
-                c.getString(R.string.risk_details_information_body_outdated_risk)
-            RiskLevelConstants.LOW_LEVEL_RISK ->
-                c.getString(
-                    if (count > 0) R.string.risk_details_information_body_low_risk_with_encounter
-                    else R.string.risk_details_information_body_low_risk
-                )
-            else -> ""
+        return when (riskState) {
+            INCREASED_RISK -> resources.getQuantityString(
+                R.plurals.risk_details_information_body_increased_risk,
+                days,
+                days
+            )
+            CALCULATION_FAILED -> c.getString(R.string.risk_details_information_body_outdated_risk)
+            LOW_LEVEL_RISK -> c.getString(
+                if (count > 0) R.string.risk_details_information_body_low_risk_with_encounter
+                else R.string.risk_details_information_body_low_risk
+            )
         }
     }
 
     /**
      * Formats the risk details text display for each risk level for the body notice
      */
-    fun getRiskDetailsRiskLevelBodyNotice(c: Context): String = when (riskLevelScore) {
-        RiskLevelConstants.INCREASED_RISK -> R.string.risk_details_information_body_notice_increased
+    fun getRiskDetailsRiskLevelBodyNotice(c: Context): String = when (riskState) {
+        INCREASED_RISK -> R.string.risk_details_information_body_notice_increased
         else -> R.string.risk_details_information_body_notice
     }.let { c.getString(it) }
 
     /**
      * Formats the risk details button display for enable tracing depending on risk level
      */
-    fun areRiskDetailsButtonsVisible(): Boolean =
-        isRiskDetailsEnableTracingButtonVisible() || isRiskDetailsUpdateButtonVisible()
-
-    /**
-     * Formats the risk details button display for enable tracing depending on risk level
-     */
-    fun isRiskDetailsEnableTracingButtonVisible(): Boolean = isTracingOffRiskLevel()
+    fun isRiskDetailsEnableTracingButtonVisible(): Boolean = isTracingOff()
 
     /**
      * Formats the risk details button display for manual updates depending on risk level and
      * background task setting
      */
-    fun isRiskDetailsUpdateButtonVisible(): Boolean = !isTracingOffRiskLevel() && isManualKeyRetrievalEnabled
+    fun isRiskDetailsUpdateButtonVisible(): Boolean = !isTracingOff() && isManualKeyRetrievalEnabled
 
```

Comment:
We could introduce a method isTracingEnabled. Negation is hard to process by the human mind

Reply:
I agree with you that negativen is to be avoided. In this instance it is more often used in the non-negated, then the negated state, so renaming it would increase negation (I did some clean up on this now though).